inference-engine/samples/speech_sample/fileutils.cpp

```diff
@@ -0,0 +1,147 @@
+            if (line.compare("BFM ") != 0) {
+                break;
+            }
+            in_file.read(reinterpret_cast<char*>(&numRows), sizeof(uint32_t));  // read number of rows
+            std::getline(in_file, line, '\4');                                  // read control-D
+            in_file.read(reinterpret_cast<char*>(&numCols), sizeof(uint32_t));  // read number of columns
+            in_file.seekg(numRows * numCols * sizeof(float), in_file.cur);      // read data
+            i++;
+        }
+        if (!in_file.eof()) {
+            std::string line;
+            std::getline(in_file, ptrName, '\0');  // read variable length name followed by space and NUL
+            std::getline(in_file, line, '\4');     // read "BFM" followed by space and control-D
+            if (line.compare("BFM ") != 0) {
+                fprintf(stderr, "Cannot find array specifier in file %s in LoadKaldiArkArray()!\n", fileName);
+                exit(-1);
+            }
+            in_file.read(reinterpret_cast<char*>(ptrNumRows), sizeof(uint32_t));     // read number of rows
+            std::getline(in_file, line, '\4');                                       // read control-D
+            in_file.read(reinterpret_cast<char*>(ptrNumColumns), sizeof(uint32_t));  // read number of columns
+            in_file.read(reinterpret_cast<char*>(&memory.front()),
+                         *ptrNumRows * *ptrNumColumns * sizeof(float));  // read array data
+        }
+        in_file.close();
+    } else {
+        fprintf(stderr, "Failed to open %s for reading in GetKaldiArkInfo()!\n", fileName);
+        exit(-1);
+    }
+
+    *ptrNumBytesPerElement = sizeof(float);
+}
+
+void ArkFile::SaveFile(const char* fileName, bool shouldAppend, std::string name, void* ptrMemory, uint32_t numRows, uint32_t numColumns) {
+    std::ios_base::openmode mode = std::ios::binary;
+    if (shouldAppend) {
+        mode |= std::ios::app;
+    }
+    std::ofstream out_file(fileName, mode);
+    if (out_file.good()) {
+        out_file.write(name.c_str(), name.length());  // write name
+        out_file.write("\0", 1);
+        out_file.write("BFM ", 4);
+        out_file.write("\4", 1);
+        out_file.write(reinterpret_cast<char*>(&numRows), sizeof(uint32_t));
+        out_file.write("\4", 1);
+        out_file.write(reinterpret_cast<char*>(&numColumns), sizeof(uint32_t));
+        out_file.write(reinterpret_cast<char*>(ptrMemory), numRows * numColumns * sizeof(float));
+        out_file.close();
+    } else {
+        throw std::runtime_error(std::string("Failed to open %s for writing in SaveKaldiArkArray()!\n") + fileName);
+    }
+}
+
+void NumpyFile::GetFileInfo(const char* fileName, uint32_t numArrayToFindSize, uint32_t* ptrNumArrays, uint32_t* ptrNumMemoryBytes) {
+    uint32_t numArrays = 0;
+    uint32_t numMemoryBytes = 0;
+
+    cnpy::npz_t my_npz1 = cnpy::npz_load(fileName);
```

Comment:
Don't we need to verify that file has been loaded here?

Reply:
No, we do not need to verify it,  Method npz_load checks file.